src/components/SearchForm/SearchForm.js

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { useQueryParam, StringParam } from 'use-query-params';
+import {
+  SearchFormWrapper,
+  Title,
+  Form,
+  Prefix,
+  Input,
+} from './SearchForm.styles';
+
+import { CTAButton } from '../CTAButton/CTAButton.styles';
+
+const SearchForm = () => {
+  const [subreddit, setSubreddit] = useQueryParam('q', StringParam);
```

Comment:
You could use a route param instead of query param to store the subreddit name like this `/search/:subreddit`. This functionality is baked in react-router natively. Then you could use react-router's useParam hook then without the need to install another package.

Reply:
At the same time, the implementation might become slightly more complicated...